### ISSUE TITLE
unresolved-symbol for non imported classes when using constructor

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1984]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
clojure-lsp suggest to add missing import class using `java-class-definition` if clj-kondo reports a class as unresolved symbol or unresolved-namespace:
```clojure
(ns foo)
(File/createTempFile ...) ;; unresolved-namespace then clojure-lsp suggests add import java.io.File 
File ;; unresolved-symbol then clojure-lsp suggests add import java.io.File 

(File. ...) ;; no clj-kondo lint report, so no clojure-lsp code action
```

As you can see, clj-kondo doesn't report for classes being initialized via constructor which makes the user don't know if that class is really working because was imported or not and if not imported, clojure-lsp doesn't suggest to add import.

**Describe the solution you'd like**
Report classes constructors that are not java.lang or clojure.lang or defrecord/deftype etc

**Describe alternatives you've considered**
Do nothing?

**Additional context**

I can help with a PR if pointed to where should be changed

**Comment by borkdude** [comment]
This isn't implemented since you can't reliably know if `File` is an existing class. Java classes can be single segment. Also you don't have to import a class to be able to invoke the constructor, e.g. `(java.io.File. ...)` is fine too.

**Comment by ericdallo** [comment]
Yeah, maybe we should check about single segment classes as a first iteration?
I know it's possible to have a class without any package but that's rare I suppose, so when we find a `Foo`, check there is no import for that and it's not a clojure lang or java lang we could warn?

**Comment by borkdude** [comment]
It used to warn but there was a false positive reported here:

https://github.com/clj-kondo/clj-kondo/issues/1362

**Comment by borkdude** [comment]
I guess clj-kondo could keep a list of known class symbols in the cache and if it doesn't it find it there, warn. This will assume you lint the whole classpath first though.

**Comment by borkdude** [comment]
I guess #1362 wasn't related.

I think we could fix this since single segment classes are indeed the exception and not recognized as a class in this case:

```
$ clj-kondo --lint - <<< 'File'
<stdin>:1:1: error: Unresolved symbol: File
linting took 8ms, errors: 1, warnings: 0
```

but we let it slide in this case:

```
borkdude@MBP2019 /tmp/inky $ clj-kondo --lint - <<< '(File.)'
linting took 10ms, errors: 0, warnings: 0
```

so we should treat it the same as in the first case.

**Comment by borkdude** [comment]
Which you already pointed out in the original description, sorry for misunderstanding the issue.

**Comment by borkdude** [comment]
Yes, PR welcome
